erpnext/hr/doctype/employee/employee.py

```diff
@@ -456,3 +394,45 @@ def has_user_permission_for_employee(user_name, employee_name):
 def get_employee_from_user(user):
 	employee_docname = frappe.db.get_value('Employee', filters={'user_id': user})
 	return employee_docname
+
+
+def send_employee_birthday_notification():
+	from frappe.email.doctype.notification.notification import create_system_notification
+
+	if not cint(frappe.db.get_single_value("HR Settings", "email_birthday_notification")):
+		return
+
+	date_today = getdate()
+	birthday_template_name = frappe.db.get_single_value("HR Settings", "default_birthday_notification_template")
+	birthday_template = frappe.get_doc("Email Template", birthday_template_name)
+
+	employee_birthday_data = frappe.db.sql("""
+		SELECT name, employee_name, personal_email, company_email
+		FROM tabEmployee
+		WHERE day(date_of_birth) = %s
+		AND month(date_of_birth)= %s
+		AND status = 'Active'
+	""", [date_today.day, date_today.month], as_dict=1)
+
+	for d in employee_birthday_data:
+		recipient = d['company_email'] or d['personal_email']
+
+		if recipient:
+			frappe.sendmail(
+				recipients=recipient,
+				subject=birthday_template.subject,
+				message=frappe.render_template(birthday_template.response, d)
+			)
+
+		else:
+			try:
+				subject = _("Birthday Emlpoyee {0} has Email Missing").format(d.employee_name)
+				create_system_notification("Employee", d.name, subject, "HR Manager")
+
+			except Exception:
+				traceback = frappe.get_traceback()
+				frappe.log_error(
+					title=_("Error: create_system_notification Failed."),
+					message=traceback,
+				)
+				frappe.db.commit()
```

Comment:
How about instead of sending a system notification for missing email, we create a big notification with list of all employees with birthdays (with mention of missing email)